spec-abstract-noclosure/typedefs.tla

```diff
@@ -0,0 +1,27 @@
+----------------------------- MODULE typedefs -----------------------------
+
+(*
+    We model two sequences of block bodies:
+
+    0 -> +1 -> +2 -> +3 -> +4 -> +5
+             \     \     \     \
+         -1 -> -2 -> -3 -> -4 -> -5
+
+    @typeAlias: body = Int;
+    @typeAlias: block = {
+        slot: Int,
+        body: $body
+    };
+    @typeAlias: checkpoint = <<$block, Int>>;
```

Comment:
If we define type checkpoint as 
`@typeAlias: checkpoint = {
        chkp_block: $block,
        chkp_slot: Int
    };`
does it have any effect to APALACHE's performance

Reply:
I would like to use records as it looks more readable and closer to the Python spec. I am sometimes confused what `vote.source[1]` and `vote.source[2]` refer to. However, this thought is subjective.

Reply:
Yes, we can refactor this specification to use records later. First, we have to make sure that it goes through all the required checks.